packages/aws-cdk-lib/aws-eks/lib/cluster.ts

```diff
@@ -1657,10 +1658,14 @@ export class Cluster extends ClusterBase {
       && this.endpointAccess._config.publicCidrs.length !== 0;
 
     // validate endpoint access configuration
+    const isLookedUpVPC: boolean = this.vpc instanceof ec2.LookedUpVpc;
+    const isSubnetIdFilter: boolean = this.vpcSubnets instanceof SubnetIdSubnetFilter;
```

Comment:
Avoid using `instanceof`, it doesn't work well in installation that contain multiple versions of the same library (it happened a lot in CDK v1, CDK v2 is less prone to that, but still).

For future reference, we do these types of checks using `Symbol`s. 

https://github.com/aws/aws-cdk/blob/2623e00546721f43a180871e2f8fb72712a9ffa1/packages/aws-cdk-lib/core/lib/stack.ts#L210-L212